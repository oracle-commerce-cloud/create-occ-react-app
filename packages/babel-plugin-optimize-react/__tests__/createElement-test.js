'use strict';

const plugin = require('../index.js');
const babel = require('@babel/core');

function transform(code) {
  return babel.transform(code, {
    plugins: [plugin],
  }).code;
}

describe('React createElement transforms', () => {
  it('should transform React.createElement calls', () => {
    const test = `
      import React from "react";

      export function MyComponent() {
        return React.createElement("div");
      }
    `;
    const output = transform(test);
    expect(output).toMatchSnapshot();
  });

  it('should transform React.createElement calls #2', () => {
    const test = `
      const React = require("react");

      export function MyComponent() {
        return React.createElement("div", null, React.createElement("span", null, "Hello world!"));
      }
    `;
    const output = transform(test);
    expect(output).toMatchSnapshot();
  });

  it('should transform React.createElement calls #3', () => {
    const test = `
      const React = require("react");

      const node = React.createElement("div", null, React.createElement("span", null, "Hello world!"));

      export function MyComponent() {
        return node;
      }
    `;
    const output = transform(test);
    expect(output).toMatchSnapshot();
  });

  it('should transform React.createElement calls #4', () => {
    const test = `
      import * as React from "react";
  
      const node = React.createElement("div", null, React.createElement("span", null, "Hello world!"));
  
      export function MyComponent() {
        return node;
      }
    `;
    const output = transform(test);
    expect(output).toMatchSnapshot();
  });
});
