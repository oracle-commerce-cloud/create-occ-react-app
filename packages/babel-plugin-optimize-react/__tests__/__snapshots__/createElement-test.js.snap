// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React createElement transforms should transform React.createElement calls #2 1`] = `
"const React = require(\\"react\\");

const __reactCreateElement__ = React.createElement;
export function MyComponent() {
  return __reactCreateElement__(\\"div\\", null, __reactCreateElement__(\\"span\\", null, \\"Hello world!\\"));
}"
`;

exports[`React createElement transforms should transform React.createElement calls #3 1`] = `
"const React = require(\\"react\\");

const __reactCreateElement__ = React.createElement;

const node = __reactCreateElement__(\\"div\\", null, __reactCreateElement__(\\"span\\", null, \\"Hello world!\\"));

export function MyComponent() {
  return node;
}"
`;

exports[`React createElement transforms should transform React.createElement calls #4 1`] = `
"import * as React from \\"react\\";
const __reactCreateElement__ = React.createElement;

const node = __reactCreateElement__(\\"div\\", null, __reactCreateElement__(\\"span\\", null, \\"Hello world!\\"));

export function MyComponent() {
  return node;
}"
`;

exports[`React createElement transforms should transform React.createElement calls 1`] = `
"import React from \\"react\\";
const __reactCreateElement__ = React.createElement;
export function MyComponent() {
  return __reactCreateElement__(\\"div\\");
}"
`;
